src/main/java/me/karakelley/http/controllers/StaticFilesController.java

```diff
@@ -27,30 +26,45 @@ public Response respond(Request request) {
 
     Response response = new Response();
     if (request.validRequestLine() && availableActions.contains(request.getMethod())) {
-      return serveDirectoryResponse(response, requestedResource);
+      return serveResponse(response, requestedResource);
     } else {
       return new InvalidRequestController().respond(request);
     }
   }
 
+  private Response serveResponse(Response response, String requestedResource) {
+    if (publicDirectory.isFile(requestedResource)) {
+      return serveFilesResponse(response, requestedResource);
+    } else {
+      return serveDirectoryResponse(response, requestedResource);
+    }
+  }
+
   private Response serveDirectoryResponse(Response response, String requestedResource) {
     response.setStatus(Status.OK);
     response.setBody(String.join("", getFilesAndDirectoryLinks(requestedResource)));
     response.setHeaders("Content-Type", "text/html");
-    response.setHeaders("Content-Length", String.valueOf(response.getBody().getBytes().length));
+    response.setHeaders("Content-Length", String.valueOf(response.getBody().length));
+    return response;
+  }
+
+  private Response serveFilesResponse(Response response, String requestedResource) {
+    response.setStatus(Status.OK);
+    response.setBody(publicDirectory.getFileContents(requestedResource));
+    response.setHeaders("Content-Type", publicDirectory.getMimeType(requestedResource));
+    response.setHeaders("Content-Length", String.valueOf(response.getBody().length));
     return response;
   }
 
   private ArrayList<String> getFilesAndDirectoryLinks(String resource) {
     List<File> files = publicDirectory.getDirectoriesAndFiles(resource);
     ArrayList<String> resources = new ArrayList<>();
     for (File file : files) {
-      if (file.isDirectory()) {
+      if (file.isDirectory() || file.isFile()) {
```

Comment:
Is anything inside the `files` list _not_ a directory or file? Do you need this check at all?

Reply:
That's a good point - At this point everything in the list is already a file or directory so I don't need to check it again